src/main/scala/com/redis/StringOperations.scala

```diff
@@ -1,26 +1,22 @@
 package com.redis
 
 import com.redis.api.StringApi
+import com.redis.api.StringApi.{Always, SetBehaviour}
 import com.redis.serialization._
 
+import scala.concurrent.duration.Duration
+
 trait StringOperations extends StringApi {
   self: Redis =>
 
-  override def set(key: Any, value: Any)(implicit format: Format): Boolean =
-    send("SET", List(key, value))(asBoolean)
-
-  @deprecated("Use the more typesafe variant", "2.14")
-  def set(key: Any, value: Any, nxxx: Any, expx: Any, time: Long): Boolean = {
-    send("SET", List(key, value, nxxx, expx, time))(asBoolean)
-  }
-
-  override def set(key: Any, value: Any, onlyIfExists: Boolean, time: SecondsOrMillis): Boolean = {
-    val nxxx = if (onlyIfExists) "XX" else "NX"
-    val expx = time match {
-      case Seconds(v) => List("EX", v)
-      case Millis(v) => List("PX", v)
+  override def set(key: Any, value: Any, whenSet: SetBehaviour = Always, expire: Duration = null)
+                  (implicit format: Format): Boolean = {
```

Comment:
This is a public API breaking change. I would provide an overload instead ..

Reply:
I was thinking about keeping old implementation or not and these are things that convinced me

1. It is not a new feature, but a cleanup. You don't need to update if you like the old API
2. The other implementations are back from 2010 - 2015. It is time to move on. 
3. It is just making more and more mess in the class and the project, I really see no point in keeping untyped version from 2013 or `SecondsOrMilliseconds` from 2015. Its 2019.

please reconsider. 
Or maybe it is worth making a branch 4.0 and we can merge it there.